### PR TITLE
Disables imag for real-valued tensors

### PR DESCRIPTION
**Pull request by mruberry**
In NumPy, calling np.imag on a real-valued tensors returns a non-writable tensor (view) of zeros. In PyTorch we don't support non-writeable tensors (or views), so we can either return a writable tensor or error. 

If we do the former, that may confuse people who try to write to the imaginary part of a real-valued tensor, and may cause a BC issue if we do support non-writable tensors. This PR errors to provide us flexibility implementation the solution we'd like in the future, while protecting users from unexpected behavior today.

### CODE REVIEW

File: aten/src/ATen/native/UnaryOps.cpp [hunk 1]
```diff
@@ -80,6 +80,11 @@ Tensor real(const Tensor& self) {
 
 Tensor imag(const Tensor& self) {
   TORCH_CHECK(!self.is_complex(), "Imag is not yet implemented for complex tensors.");
+  TORCH_CHECK(self.is_complex(),
```
Comment (gchanan) [inline aten/src/ATen/native/UnaryOps.cpp]: should we just...delete it?  It's kind of mean to tell people it must be a complex tensor but then if they pass a complex tensor that it's not implemented.  Sometimes that's unavoidable (e.g. if the errors are far apart), but it seems a bit silly here.

Alternatively, write a global error message?  Just something like "imag is not currently supported".

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -2491,20 +2486,15 @@ def merge_dicts(*dicts):
 imag(input, out=None) -> Tensor
 
 Returns the imaginary values of the given :attr:`input` tensor. If
-:attr:`input` is a real (non-complex) tensor this is equivalent to
-``torch.zeros_like(input)``.
+:attr:`input` is a real (non-complex) tensor it has no imaginary part
```
Comment (gchanan) [inline torch/_torch_docs.py]: the line above also seems out of date.

Comment (mruberry) [reply torch/_torch_docs.py]: Take a look. I added a warning to try and clarify this while describing the intent of the function generically.

File: aten/src/ATen/native/UnaryOps.cpp [hunk 3]
```diff
@@ -79,7 +79,9 @@ Tensor real(const Tensor& self) {
 }
 
 Tensor imag(const Tensor& self) {
-  TORCH_CHECK(!self.is_complex(), "Imag is not yet implemented for complex tensors.");
+  TORCH_CHECK(false, "Imag is not yet implemented.");
```
Comment (mruberry) [inline aten/src/ATen/native/UnaryOps.cpp]: Imag -> imag

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -1019,7 +1019,7 @@ def merge_dicts(*dicts):
            r"""
 real(input, out=None) -> Tensor
 
-Returns the real values of the given :attr:`input` tensor. If
+Returns the real part of the :attr:`input` tensor. If
 :attr:`input` is a real (non-complex) tensor, this function just
 returns it.
```
Comment (mruberry) [inline torch/_torch_docs.py]: Add warning: not yet implemented for complex tensors.

**Review by gchanan — APPROVED** [review]
only minor changes left

**Pull request closed without merging**